LocalKeyClientEncryption/LocalKeyClientEncryption.cs

```diff
@@ -0,0 +1,309 @@
+        * MUST MATCH PARAMETERS USED TO ENCRYPT CLIENT SIDE ENCRYPTED DATA
+        */
+        static void Main()
+        {
+            var config = ConfigurationManager.AppSettings;
+
+            //Credentials of Service Principal
+            TokenCredential credential =
+                new ClientSecretCredential(
+                    config["tenantId"],
+                    config["clientId"],
+                    config["clientSecret"]
+                    );
+
+            string keyVaultName = config["keyVaultName"];
+            string connectionString = config["connectionString"];
+
+
+            //Edit the Following as Needed
+            //Program Creates the Following using the Provided Names
+            string containerName = "example" + Guid.NewGuid().ToString();
+            string fileName = "example.txt";
+            string keyVaultKeyName = "testKey";
+            string encryptionScopeName = "myencryption";
+            string customerProvidedKey = "dfD3Jb#6htqfpoj@gGpomDAv21035%21";     //Key used for Customer Provided Key Server Side Encryption
+
+            byte[] localKeyBytes = ASCIIEncoding.UTF8.GetBytes(customerProvidedKey);
+
+
+
+
+            //File Path for local file used to upload and reupload
+            string localPath = "./data/";
+            Directory.CreateDirectory(localPath);
+            string localFilePath = Path.Combine(localPath, fileName);
+
+            //Creating Key Encryption Key object for Client Side Encryption
+            SampleKeyEncryptionKey keyEncryption = new SampleKeyEncryptionKey(config["clientSideCustomerProvidedKey"]);
+
+            //Set up Client Side Encryption Options used for Client Side Encryption
+            ClientSideEncryptionOptions clientSideOptions = new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+            {
+                KeyEncryptionKey = keyEncryption,
+                KeyWrapAlgorithm = config["keyWrapAlgorithm"]
+            };
+
+            //Create Blob Service Client
+            BlobServiceClient blobServiceClient = new BlobServiceClient(connectionString);
+
+            //Run Setup Function that creates and example container and blob
+            SetupForExample(blobServiceClient, containerName, fileName, localFilePath, clientSideOptions);
+
+            Console.Write("Press Enter to change to SSE");
+            Console.ReadLine();
+
+            //Convert Client Side Encryption Blob to Server Side Encrytion with Microsoft Managed Keys, Customer Managed Keys, and Customer Provided Keys
+            CSEtoSSE(connectionString, containerName, fileName, localFilePath, clientSideOptions, encryptionScopeName, keyVaultName, keyVaultKeyName, credential, localKeyBytes);
+
+            //CLEAN UP EXAMPLE CONTAINER AND FILES
+            Console.Write("Press Enter to begin clean up");
+            Console.ReadLine();
+
+            //Delete downloaded files
+            CleanUp();
+
+            Console.WriteLine("Complete");

```

Comment:
Maybe change this from printing out "Complete" to "Completed each type of encryption"